```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':fineract-core:spotlessJavaCheck'.
> The following files had format violations:
      src/test/java/org/apache/fineract/portfolio/savings/domain/SavingsHelperNPExceptionTest.java
          @@ -23,11 +23,12 @@
           import·java.time.LocalDate;
           import·java.util.ArrayList;
           import·java.util.List;
          -import·org.apache.fineract.infrastructure.core.domain.LocalDateInterval;·//·Ensure·this·import·exists
          +import·org.apache.fineract.infrastructure.core.domain.LocalDateInterval;
           import·org.apache.fineract.portfolio.savings.SavingsPostingInterestPeriodType;
           import·org.junit.jupiter.api.Test;
           
           class·SavingsHelperNPExceptionTest·{
          +
           ····private·final·SavingsHelper·savingsHelper·=·new·SavingsHelper(null);

           
           ····@Test
          @@ -41,13 +42,8 @@
           ········Integer·financialYearBeginningMonth·=·1;
           ········List<LocalDate>·postInterestAsOn·=·new·ArrayList<>();
           
          -········List<LocalDateInterval>·result·=·savingsHelper.determineInterestPostingPeriods(
          -················start,
          -················end,
          -················nullPostingPeriodType,
          -················financialYearBeginningMonth,
          -················postInterestAsOn
          -········);
          +········List<LocalDateInterval>·result·=·savingsHelper.determineInterestPostingPeriods(start,·end,·nullPostingPeriodType,
          +················financialYearBeginningMonth,·postInterestAsOn);
           
           ········assertTrue(result.isEmpty(),·"Result·should·be·empty·when·postingPeriodType·is·null");
           ····}
  Run './gradlew :fineract-core:spotlessApply' to fix these violations.

--- a/fineract-core/src/test/java/org/apache/fineract/portfolio/savings/domain/SavingsHelperNPExceptionTest.java
+++ b/fineract-core/src/test/java/org/apache/fineract/portfolio/savings/domain/SavingsHelperNPExceptionTest.java
@@ -28,6 +28,7 @@ import org.apache.fineract.portfolio.savings.SavingsPostingInterestPeriodType;
 import org.junit.jupiter.api.Test;
 
 class SavingsHelperNPExceptionTest {
+
     private final SavingsHelper savingsHelper = new SavingsHelper(null);
 
     @Test

--- a/fineract-core/src/test/java/org/apache/fineract/portfolio/savings/domain/SavingsHelperNPExceptionTest.java
+++ b/fineract-core/src/test/java/org/apache/fineract/portfolio/savings/domain/SavingsHelperNPExceptionTest.java
@@ -28,22 +28,18 @@ import org.apache.fineract.portfolio.savings.SavingsPostingInterestPeriodType;
 import org.junit.jupiter.api.Test;
 
 class SavingsHelperNPExceptionTest {
-    
     private final SavingsHelper savingsHelper = new SavingsHelper(null);
 
     @Test
     void shouldNotThrowNullPointerExceptionWhenPostingPeriodTypeIsNull() {
         LocalDate start = LocalDate.of(2025, 1, 1);
         LocalDate end = LocalDate.of(2025, 4, 1);
-        
         SavingsPostingInterestPeriodType nullPostingPeriodType = null;
 
         Integer financialYearBeginningMonth = 1;
         List<LocalDate> postInterestAsOn = new ArrayList<>();
-        
         List<LocalDateInterval> result = savingsHelper.determineInterestPostingPeriods(start, end, nullPostingPeriodType,
                 financialYearBeginningMonth, postInterestAsOn);
-                
         assertTrue(result.isEmpty(), "Result should be empty when postingPeriodType is null");
     }
 }

--- a/fineract-core/src/test/java/org/apache/fineract/portfolio/savings/domain/SavingsHelperNPExceptionTest.java
+++ b/fineract-core/src/test/java/org/apache/fineract/portfolio/savings/domain/SavingsHelperNPExceptionTest.java
@@ -23,32 +23,27 @@ import static org.junit.jupiter.api.Assertions.assertTrue;
 import java.time.LocalDate;
 import java.util.ArrayList;
 import java.util.List;
-import org.apache.fineract.infrastructure.core.domain.LocalDateInterval; // Ensure this import exists
+import org.apache.fineract.infrastructure.core.domain.LocalDateInterval;
 import org.apache.fineract.portfolio.savings.SavingsPostingInterestPeriodType;
 import org.junit.jupiter.api.Test;
 
 class SavingsHelperNPExceptionTest {
+    
     private final SavingsHelper savingsHelper = new SavingsHelper(null);
 
     @Test
     void shouldNotThrowNullPointerExceptionWhenPostingPeriodTypeIsNull() {
         LocalDate start = LocalDate.of(2025, 1, 1);
         LocalDate end = LocalDate.of(2025, 4, 1);
-
-        // This is the NULL that was causing the crash
+        
         SavingsPostingInterestPeriodType nullPostingPeriodType = null;
 
         Integer financialYearBeginningMonth = 1;
         List<LocalDate> postInterestAsOn = new ArrayList<>();
-
-        List<LocalDateInterval> result = savingsHelper.determineInterestPostingPeriods(
-                start,
-                end,
-                nullPostingPeriodType,
-                financialYearBeginningMonth,
-                postInterestAsOn
-        );
-
+        
+        List<LocalDateInterval> result = savingsHelper.determineInterestPostingPeriods(start, end, nullPostingPeriodType,
+                financialYearBeginningMonth, postInterestAsOn);
+                
         assertTrue(result.isEmpty(), "Result should be empty when postingPeriodType is null");
     }
 }

